### DISCUSSION TITLE
what's the mdx language server status by far?

### DISCUSSION THREAD

**Discussion by HomyeeKing** [mdx-js/mdx#2269] [Q&A]
I intend to write `tsx` code block at `.mdx` and I need to get IntelliSense while using some components

```tsx
// comp.tsx
export default ({title}:{title:string})=><div>{title}</div>
```

```mdx
import Comp from './comp' 

// will remind me to add title prop
<Comp  />
```

I've installed vscode-mdx extension and enable `mdx.experimentalLanguageServer`, noting happens

**Comment by remcohaszing** [comment]
I tried your code and it works for me. When hovering over `Comp` in either the import or the JSX element, it shows hover information. Is this not the case for you? What did you expect to happen?

---

To answer the more generic question in the title: The status is that the language server works quite well IMO, but new issues are still surfacing, so I prefer to keep it behing a flag. Also the language server hasn’t been released to npm yet, so this experience is still only available in VSCode. I’m also talking to the Volar team to streamline the IntelliSense experience for MDX, Vue, and other similar languages that embed JavaScript.

**Comment by HomyeeKing** [comment]
@remcohaszing  thanks for your response. it does have IntelliSense while inputting props, but no type check

also, I would like it works in code block too


<pre>
// foo.mdx
<code>
```tsx
 // some tsx code here also have intellisense and  type check
```
</code>
</pre>

**Reply by remcohaszing** [reply]
What type checking are you missing? Your example shows a React component with no type annotations. This means the props are of type `any`.

re: code blocks: There’s an open issue for that: https://github.com/mdx-js/mdx-analyzer/issues/287

**Reply by HomyeeKing** [reply]
![image](https://user-images.githubusercontent.com/49113249/224976673-d15185b2-0ac9-46c7-9916-a5bb96110225.png)

still no type check

**Reply by remcohaszing** [reply]
After processing, `.mdx` files are treated as `.jsx` files by the underlying TypeScript language service. To get this marked as an error, you should enable `checkJs` in your `tsconfig.json`.

**Reply by HomyeeKing** [reply]
I've enabled `checkJs`, but still the same
![image](https://user-images.githubusercontent.com/49113249/225186772-c1d38847-74aa-435a-9794-56c53f3c3be4.png)

by the way, I just enabled `mdx.experimentalLanguageServer` and not config the `eslint-mdx-*` with the extension, should I do that?

**Reply by remcohaszing** [reply]
I am not able to reproduce this issue. Is the MDX file be excluded by your include / exclude configuration? Could you create a minimal reproduction?

**Reply by remcohaszing** [reply]
And ESLint is unrelated to this. I know it’s mentioned in the readme, so I understand the confusion.

**Reply by HomyeeKing** [reply]
https://github.com/HomyeeKing/reproduce-mdx-type-check

here is the repo

**Reply by remcohaszing** [reply]
It’s actually working for me.

![image](https://user-images.githubusercontent.com/779047/225319108-8a3d060c-8a45-412c-a837-b9e2ed106b0a.png)

Are you using an old version of the MDX extension?

**Reply by HomyeeKing** [reply]
emmm...   it suddenly works now  and I have no idea why 🤷‍♂️  
anyway thanks for your time!
